---
title: "MICB425 Portfolio - Microbial Ecological Genomics: From Genomes to Biomes, Spring 2018"
author: "Student First Name, Preferred Name, Last Name, Student ID"
date: "version `r format(Sys.time(), '%B %d, %Y')`"
output: 
  html_document:
    toc: true
---
# Module 01
Reserve the first level headings (#) for the start of a new Module. This will help to organize your portfolio in an intuitive fashion.  
**Note: Please edit this template to your heart's content. This is meant to be the armature upon which you build your individual portfolio. You do not need to keep this instructive text in your final portfolio, although you do need to keep module and assignment names so we can identify what is what.**


## Module 01 portfolio check
The first of your second level headers (##) is to be used for the portfolio content checks. The Module 01 portfolio check has been built for you directly into this template, but will also be available as a stand-alone markdown document available on the MICB425 GitHub so that you know what is required in each module section in your portfolio. The completion status and comments will be filled in by the instructors during portfolio checks when your current portfolios are pulled from GitHub.

* Installation check
    + Completion status:
    + Comments:
* Portfolio repo setup
    + Completion status: 
    + Comments:
* RMarkdown Pretty html Challenge
    + Completion status:
    + Comments:
* Evidence worksheet_01
    + Completion status:
    + Comments:
* Evidence worksheet_02
    + Completion status:
    + Comments:
* Evidence worksheet_03
    + Completion status:
    + Comments:
* Problem Set_01
    + Completion status:
    + Comments:
* Problem Set_02
    + Completion status:
    + Comments:
* Writing assessment_01
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments  
    
## Data science Friday  
The remaining second level headers (##) are for separating data science Friday, regular course, and project content. In this module, you will only need to include data science Friday and regular course content; projects will come later in the course.

### Installation check
Third level headers (###) should be used for links to assignments, evidence worksheets, problem sets, and readings, as seen here.  

Use this space to include your installation screenshots.

![rmarkdown screenshot](/Users/victoriapanwala/Documents/MICB425_portfolio/Screen Shot 2018-01-10 at 7.44.44 PM.png)
![github screenshot](/Users/victoriapanwala/Documents/MICB425_portfolio/Screen Shot 2018-01-10 at 7.56.03 PM.png)
![terminal setup](Screen Shot 2018-01-10 at 7.48.48 PM.png)

### Portfolio repo setup
Detail the code you used to create, initialize, and push your portfolio repo to GitHub. This will be helpful as you will need to repeat many of these steps to update your porfolio throughout the course.



### RMarkdown pretty html challenge
Paste your code from the in-class activity of recreating the example html.

---
title: "Pretty_html"
author: "Victoria Panwala (14028147)"
date: "1/21/2018"
output: 
  html_document:
    toc: true
---
version January 18, 2018

## R Markdown PDF Challenge

The following assignment is an exercise for the reproduction of this .html document using the RStudio and RMarkdown tools we’ve shown you in class. Hopefully by the end of this, you won’t feel at all the way this poor PhD student does. We’re here to help, and when it comes to R, the internet is a really valuable resource. This open-source program has all kinds of tutorials online.

![http://phdcomics.com/ Comic posted 1-17-2018](http://phdcomics.com/comics/archive/phd011718s.gif)

## Challenge Goals
The goal of this R Markdown html challenge is to give you an opportunity to play with a bunch of different RMarkdown formatting. Consider it a chance to flex your RMarkdown muscles. Your goal is to write your own RMarkdown that rebuilds this html document as close to the original as possible. So, yes, this means you get to copy my irreverant tone exactly in your own Markdowns. It’s a little window into my psyche. Enjoy =)

hint: go to the PhD Comics website to see if you can find the image above
If you can’t find that exact image, just find a comparable image from the PhD Comics website and include it in your markdown

### Here's a header!
Let’s be honest, this header is a little arbitrary. But show me that you can reproduce headers with different levels please. This is a level 3 header, for your reference (you can most easily tell this from the table of contents).

#### Another header, now with maths
Perhaps you’re already really confused by the whole markdown thing. Maybe you’re so confused that you’ve forgotton how to add. Never fear! ~~A calculator~~ R is here:

```{r}
1231521+12341556280987
```

### Table Time
Or maybe, after you’ve added those numbers, you feel like it’s about time for a table!
I’m going to leave all the guts of the coding here so you can see how libraries (R packages) are loaded into R (more on that later). It’s not terribly pretty, but it hints at how R works and how you will use it in the future. The summary function used below is a nice data exploration function that you may use in the ^future^.

```{r}
library(knitr)
kable(summary(cars),caption="I made this table with kable in the knitr package library")
```

And now you’ve almost finished your first RMarkdown! Feeling excited? We are! In fact, we’re so excited that maybe we need a big finale eh? Here’s ours! Include a fun gif of your choice!

![can this do gifs?](https://media.giphy.com/media/3o7TKxlVXJsQsDX2og/giphy.gif))

## Origins and Earth Systems
### Evidence worksheet 01 
The template for the first Evidence Worksheet has been included here. The first thing for any assignment should link(s) to any relevant literature (which should be included as full citations in a module references section below).

You can copy-paste in the answers you recorded when working through the evidence worksheet into this portfolio template. 

As you include Evidence worksheets and Problem sets in the future, ensure that you delineate Questions/Learning Objectives/etc. by using headers that are 4th level and greater. This will still create header markings when you render (knit) the document, but will exclude these levels from the Table of Contents. That's a good thing. You don't' want to clutter the Table of Contents too much.

[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

#### Learning objectives
Describe the numerical abundance of microbial life in relation to ecology and biogeochemistry of Earth systems. 

#### General questions
#####* What were the main questions being asked?  

How do we estimate the prokaryotic population of the world? And what is it made up of? 

What are the uncertainties that come with this measurement?

Which environments contain the most prokaryotic biomass?

How does this biomass affect global nutrient cycles? (e.g. P, C, N) 

#####* What were the primary methodological approaches used?  

Prokaryotic estimates were based upon average data from the following four environments: aquatic environments, soil, subsurface, and “other habitats” including in or on animal or plant surfaces or in the air. They used experimentally derived values to perform these calculations, but interestingly, not the same value sets for each environment. For example, some calculations included cell volume, while others included just the area of the environment. Vi 


Also, they compared their calculated values with some from other papers, which resulted in some differences that they attempted to explain. 

#####* Summarize the main results or findings.  

They found that prokaryotes contain about half of the organic carbon on earth, and 90% of the nutrients (compared to plants) 
In brief, the prokaryotic biomass and thus their contribution to global cycles is very large - doubling estimates of the amount of carbon stored in living organisms globally. They broke down the calculations into four environments: aquatic environments, soil, subsurface, and “other habitats”. 


Aquatic environments- this includes the open ocean, sediment in the ocean, freshwater and saline lakes (3 orders of magnitude less) and polar regions. Prokaryotes are ubiquitous in these environments - 1180 x 10^26^ cells. 

Soil- surprisingly, there are less prokaryotes in forest soils than in other soils. The estimates varied by ecosystem. 255.6 x 10^27^ cells.

Subsurface- e.g. terrestrial habitats below 8 m and marine sediments below 10 cm. (this includes groundwater too) This environment is difficult to estimate because it is difficult to obtain uncontaminated samples. However, it has been suggested to be enormous. 3.8 x 10^30^ cells.

Other environments - discussed the prokaryotes that live on animals, insects, and plants, and also those in the air/atmosphere. 53.024 x 10^23^ cells (several orders of magnitude smaller) 

These large numbers mean that not only carbon, but N and P are stored in globally significant amounts in prokaryotes. 

Disproves Kluyver’s estimate that 1/2 of the living protoplasm on earth is microbial - likely this number is far too conservative. The paper also discusses growth rates to estimate cell turnover, and fluxes in and out of these environments.  

#####* Do new questions arise from the results?  

In subsurface environments, the turnover time of cells seems exceedingly large, is this a good estimate?

Where does the energy in the subsurface environments come from? Photosynthesis? Chemolithotrophy?

From the passage “in the polar regions, a relatively dense community of algae and prokaryotes forms at the water-ice interface” - why does this occur?

How accurate can these calculations be if they are based upon just a few estimates?

How much flux occurs among all of these prokaryotic environments? Especially the subsurface environment, if so many cells are hypothesized to be metabolically inactive, how much flux can occur? Is it more of a pool than a flux? 

#####* Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  

The estimates described in the paper introduce a large amount of uncertainty because no matter how many samples you collect, you are still having to generalize this data for the entire earth. You cannot possibly collect enough data to have any degree of accuracy in your prediction. However, otherwise, their experimental logic made sense.

Lastly, the estimates for each environment were likely collected using different methods- and by different people. Therefore, each method probably has its own pros and cons, and contributes its own level of uncertainty to the proceeding estimates.   

### Problem set 01
#### Learning objectives: 
Describe the numerical abundance of microbial life in relation to the ecology and biogeochemistry of Earth systems.

#### Specific questions:
#####* What are the primary prokaryotic habitats on Earth and how do they vary with respect to their capacity to support life? Provide a breakdown of total cell abundance for each primary habitat from the tables provided in the text.  

The primary prokaryotic habitats on earth are split into aquatic habitats, soil, and subsurface habitats. According to table 5 of the text, there are 12 x 10^28^ cells in aquataic habitats, 26 x 10^28^ cells in the soil, and interestingly, 355 x 10^28^ and 25-250 x 10^28^ prokaryotic cells in the oceanic and terrestrial subsurface respectively. However, in order to rank these habitats based on their capacity to support life, we must come up with a universal definition for "capacity to support life". If you were to define this as the total number of prokaryotic cells in a given habitat, it would appear that the oceanic subsurface habitat has the greatest capacity to support life. However, this does not take into account whether these cells are metabolically active, or their turnover time, or the total area occupied by the habitat.  

#####* What is the estimated prokaryotic cell abundance in the upper 200 m of the ocean and what fraction of this biomass is represented by marine cyanobacterium including Prochlorococcus? What is the significance of this ratio with respect to carbon cycling in the ocean and the atmospheric composition of the Earth?  
2.8 x 10^28^ cells in the upper 200m 

The average density is 5 x 10^5^ cells/mL 

To calculate what fraction of this ratio are cyanobacteria: 

4 x10^4^ cells/ml / 5 x 10^5^ cells/ml x 100 = 8%

This ratio is significant because these cells are autotrophs, which means that they are responsible for asimilating inorganic carbon into this environment, and thus are an important aspect of carbon cycling in the ocean. This is not only important for aquatic environments, but for the atmospheric composition of the earth as well. This is because some organic carbon fixed by these autotrophs are not respired and stored in marine sediment. Since respiring this material generally requires oxygen, its long-tem storage means that oxygen can remain in significant levels in the earth's atmosphere. This is in contrast to terrestrial systems, where carbon fixed by autotrophs is generally respired. 

#####* What is the difference between an autotroph, heterotroph, and a lithotroph based on information provided in the text?  

Autotroph - produces organic complex carbons from simple inorganic substances such as carbon dioxide.
Heterotroph - takes up organic carbon to produce energy and synthesize compounds 
Lithotroph - uses an inorganic substrate to obtain reducing equivalents for use in biosynthesis or energy conservation via aerobic or anaerobic respiration

#####* Based on information provided in the text and your knowledge of geography what is the deepest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this depth?  

Temperature is the limiting factor in subsurface environments, and at around 4 km in terestrial environments the temperature reaches 125 degrees celsius. This is the generally agreed upon temperature limit of prokaryotic life.

The deepest habitat capable of supporting prokaryotic life is the Mariana Trench, which is 10.9 km deep, then cellular life should be able to persist another 4 km deeper in the subsurface- so 14.9 km total.

#####* Based on information provided in the text your knowledge of geography what is the highest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this height?  

Mount everest is 8.8 km high, so that would be the highest terestrial habitat capable of supporting prokaryotic life.

Additionally, in the text, bacteria in the air were discussed. However, are these bacteria actually metabolically active? They could just be spore formers, or metabolically inactive until they reach an environment that is more viable. 

The paper stated 77 km, but this does not seem very realistic, because the limiting factors in these environments include nutrient availability, UV radiation, and temperature.
So I would say more like 20 km high.

#####* Based on estimates of prokaryotic habitat limitation, what is the vertical distance of the Earth’s biosphere measured in km?  

I would say that the vertical distance of the Earth's biosphere is a range of 24 km - 34 km (due to the Mariana trench)

#####* How was annual cellular production of prokaryotes described in Table 7 column four determined? (Provide an example of the calculation)  

Population size x turnovers/year = cells/year

Marine heterotrophs:
3.6 x 10^28^ cells x 365 days / 16 days/turnover = 8.2 x 10^29^ cells/year

#####* What is the relationship between carbon content, carbon assimilation efficiency and turnover rates in the upper 200m of the ocean? Why does this vary with depth in the ocean and between terrestrial and marine habitats?  

Assuming the carbon efficiency is 20% 
If there is around 5-20 fg of carbon in a prokaryotic cell, 20 fg C/cell = 20^-30^ Pg/cell

3.6 x 10^28^ cells x 20^-30^ Pg/cell = 0.72 Pg C are trapped in marine heterotrophs
To calculate the total carbon flux, we should multiply this value by 5, but the authors used 4, so 4 x .72 = 2.88 Pg/year

51 Pg C/year, 85% of the carbon in the photic zone is consumed = 43 Pg C 

43 Pg C/year / 2.88 Pg/year = 14.4 turnovers/year 
1 turnover every 25.4 days

This varies with depth in the ocean due to access to sunlight, as photosynthesis provides the energy necessary for carbon fixation. In terestrial habitats, a number of factors including differences in depth, sediment, nutrient availability, and cell density contribute to the differences in carbon fixation and turnover rate.

#####* How were the frequency numbers for four simultaneous mutations in shared genes determined for marine heterotrophs and marine autotrophs given an average mutation rate of 4 x 10^-7^ per DNA replication? (Provide an example of the calculation with units. Hint: cell and generation cancel out)  

In table 7 of the paper, the turnover rate was discussed for each habitat, something that is essential for calculating the mutation frequency. 

To calculate the frequency of four simultaneous mutations, (4x10^-7^ per gene per DNA replication)^4^=2.56 x 10^-26^ occurences per gene per DNA replication, in other words mutations per generation

Then we calculate the cell turnover, or the number of DNA replications that occur per hour for all the cells in the environment

3.6 x 10^28^ cells x 22.8 turnovers/year = 8.2 x 10^29^ cells/year 
365 days / 16 days/turnover = 22.8 turnovers/year

Note: after this when I use the word mutation, this actually means the occurence of four simultaneous mutations

8.2 x 10^29^ cells/year / 2.56 x 10^-26^ mutations per generation = 2.1 x 10^4^ mutations/year

2.1 x 10^4^ mutations/year x 1 year/365 days = 57.53 mutations/day 
57.53 mutations/day x 1 day/24 hours = 2.39 mutations/hour

In other words (the inverse of this ratio), 0.4 hours per four simultaneous mutation event. 

#####* Given the large population size and high mutation rate of prokaryotic cells, what are the implications with respect to genetic diversity and adaptive potential? Are point mutations the only way in which microbial genomes diversify and adapt?  

Given the large population size and high mutation rate of prokaryotic cells, this indicates that prokaryotic cells have a very large adaptive potential. As prokaryotes have existed on the earth for billions of years, this corresponds to an incredibly large amount of genetic diversity.

However, it is foolhardy to assume that point mutations are the only way that microbial genomes diversify and adapt. Infection by bacteriophages can transport foreign DNA into a cell, some cells can share plasmids using conjugative pili, and others can uptake exogenous DNA from their environment. Additionally, other mutation events can occur independent of point mutations in a single cell, such as gene duplication or deletion. 

##### * What relationships can be inferred between prokaryotic abundance, diversity, and metabolic potential based on the information provided in the text? 

Based on the information provided in the text, I would say that prokaryotic abundance and metabolic potential is highly related to the diversity of organisms present in the biosphere. Given the large abundance of prokaryotic life, and its correspondingly large mutation rate, over the course of geologic time this has generated an incredibly diverse set of metabolic capabilities of individual prokaryotes. These different metabolic abilities have enabled prokaryotes to colonize the entire biosphere (24-34 km of the earth's surface, subsurface, and lower atmosphere).

### Module 01 references
Utilize this space to include a bibliography of any literature you want associated with this module. We recommend keeping this as the final header under each module.

An example for Whitman and Wiebe (1998) has been included below.

Whitman WB, Coleman DC, and Wiebe WJ. 1998. Prokaryotes: The unseen majority. Proc Natl Acad Sci USA. 95(12):6578–6583.  [PMC33863](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)