---
title: "Module 3 Portfolio"
author: "Victoria Panwala (14028147)"
date: "3/19/2018"
output: 
  html_document:
    toc: true
---

Module 3 Microbial Species Concepts

#Evidence worksheet_05 “Extensive mosaic structure”

General Questions:

#####• What were the main questions being asked?

How different are bacteria (even those that technically belong to the same species) genetically?
What defines a species? 
What proteins are shared between these three strains of E. coli? Can we use this information to infer a bit about the evolutionary history of each strain?
What about island genes and horizontal transfer? Can we use the different codon usage patterns of island genes to infer whether the island genes were horizontally transfered?

#####• What were the primary methodological approaches used?

Essentially just cloning, sequencing, and then sequence analysis and annotation. 
They created whole genome libraries from genomic DNA of the three strains and then sequenced the clones. They used the programs MAGPIE and GLIMMER to annotate the genome and find the ORFs, and then used BLAST to find the predicted protein products. 
They also analyzed different codon usage patterns to identify island genes gained by horizontal gene tranfer (a long time ago).

#####• Summarize the main results or findings. 

The considerable variation in these E. coli lineages or strains to allow them to occupy different ecological niches. This indicates that these extraintestinal E. coli have evolved fairly independently from one another.

Interestingly, there are some "universal insertion targets" in the E. coli genome, that are the site for insertion of DNA from horizontal gene transfer (even though the genes transferred differ between the strains) for all of the strains surveyed. Specifically, it is interesting that these sites are more likely to incorporate foreign DNA than elsewhere in the genome.
 
#####• Do new questions arise from the results?

How do we define E. coli as a species? Does the fact that these organisms have essentially evolved to occupy different biological niches indicate that they should be characterized as different species?

#####• Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 
 
Part 2: Learning objectives: 

#####•	Comment on the creative tension between gene loss, duplication and acquisition as it relates to microbial genome evolution

The gain and loss of genes, and the diversity that arises from this paradigm is mediated by natural selection. 
If you don't aquire specialized genes to survive in diverse environments, you will not be able to survive period. This both mediates gain of unique genetic elements, and loss of genetic elements no longer required for survival in said environment: genetic dead weight.

#####•	Identify common molecular signatures used to infer genomic identity and cohesion

Some common signature they used in the paper was type III secretion systems, partial prophage genomes, fimbrial adhesins, iron sequestration stems, autotransporters, and phase-switch recombinases. They also used patterns of island appearance vs backbone DNA (identified using different codon usage patterns) too.

#####•	Differentiate between mobile elements and different modes of gene transfer

#####Based on your reading and discussion notes, explain the meaning and content of the following figure derived from the comparative genomic analysis of three E. coli genomes by Welch et al. Remember that CFT073 is a uropathogenic strain and that EDL933 is an enterohemorrhagic strain. Explain how this study relates to your understanding of ecotype diversity. Provide a definition of ecotype in the context of the human body. Explain why certain subsets of genes in CFT073 provide adaptive traits under your ecological model and speculate on their mode of vertical descent or gene transfer.

An ecotype is a distinct form of a species that occupies a specific niche. For example, in the context of E. coli and the human body, enterohemorrhagic and uropathogenic strains of E. coli are different ecotypes because they inhabit different "environments" or niches in the human body (urinary tract vs gastrointestinal system). 

In the diagram, CFT073 appears to possess a variety of pap proteins that the enterohemorrhagic strain EDL933 does not posess. This is likely because pap proteins are used to assemble a pilus necessary for attatchment of E. coli to host cell surfaces. This pilus is necessary in uropathogenic e coli like CFT073 because in order to cause infection, they must be resistant to mechanical force from the urine stream. Enterohemorragic e. coli do not necesarily require this "sticky" phenotype in order to cause infection. Therefore, I would hypothesize that these pap genes were obtained during a horizontal gene transfer event early on in the evolutionary history of the CFT073 strain, and would be common to other strains of uropathogenic e. coli. That, or the pap pilus is a backbone gene that has been deleted from the EDL933 strain genome due to lack of use. 

#Problem set_04 “Fine-scale phylogenetic architecture”
```{r}
#To make tables
library(kableExtra)
library(knitr)
#To manipulate and plot data
library(tidyverse)
```
```{r}
sample_data1 = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14),
  name = c("vine", "bricks", "skittles", "mike & ikes", "gummy bears", "M & Ms", "Hershey Kisses", "Sour bear", "Sour fruit", "Sour hexa", "Sour bottle", "Sour swirl", "Jujubes", "wine candy"),
  characteristics = c("red vines", "candy lego bricks", "not m and ms", "mike and ikes", "bear shaped", "not skittles", "foil wrapped", "bear shaped and sour", "sour and a fruit", "sour and hexagon shaped", "sour and bottle shaped", "sour and swirly", "honestly not sure what these look like", "wine shaped but not alcoholic"),
  occurences = c(14, 18, 187, 174, 101, 241, 16, 3, 2, 6, 3, 3, 24, 9)
)
```
####This data is from Nishi, Jack, John, Leilnaz, and Helen's group - they provided me with the species count data. 
```{r}
sample_data1 %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)
```
No matter how many samples you take, you can never say for certain that you have collected a sample that accurately portrays the diversity of the environment. All you can do is reduce the chance that you have taken an unrepresentative sample, by taking multiple (large) samples. Given the size of the metagenomic data set, I would say that the majority of different species possible were sampled. 

####Part 2: collector's curve
```{r}
sample_data2 = data.frame(
  x = c(1,2,3,4,5,6,7,8,9,10),
  y = c(1,2,3,4,4,5,5,5,6,6)
)
```
```{r}
ggplot(sample_data2, aes(x=x, y=y)) +
  geom_point() +
  geom_smooth() +
  labs(x="Cumulative number of individuals classified", y="Cumulative number of species observed")
```
Note: did not have the data for this portion. I just included the code so that I would have it for future reference. However, I would say that if the curve flattens out, you have taken a large enough sample of your environment for it to be fairly representative of said environment. 

####Part 3: Diversity Estimates
Diversity: Simpson Reciprocal index calculation 
Simpson reciprocal index for total community.
```{r}
species1 = 14/(736)
species2 = 18/(736)
species3 = 187/(736)
species4 = 174/(736)
species5 = 101/(736)
species6 = 241/(736)
species7 = 16/(736)
species8 = 3/(736)
species9 = 2/(736)
species10 = 6/(736)
species11 = 3/(736)
species12 = 3/(736)
species13 = 24/(736)
species14 = 9/(736)

1 / (species1^2 + species2^2 + species3^2 + species4^2 + species5^2 + species6^2 + species7^2 + species8^2 + species9^2 + species10^2 + species11^2 + species12^2 + species13^2 + species14^2)
```
Simpson Reciprocal index for a smaller sample of that community: (sample)
```{r}
species1 = 2/(153)
species2 = 5/(153)
species3 = 37/(153)
species4 = 30/(153)
species5 = 19/(153)
species6 = 64/(153)
species7 = 2/(153)
species8 = 0/(153)
species9 = 1/(153)
species10 = 0/(153)
species11 = 0/(153)
species12 = 0/(153)
species13 = 8/(153)
species14 = 3/(153)

1 / (species1^2 + species2^2 + species3^2 + species4^2 + species5^2 + species6^2 + species7^2 + species8^2 + species9^2 + species10^2 + species11^2 + species12^2 + species13^2 + species14^2)
```



Richness: Chao1 richness estimator of entire community
```{r}
14 + 0^2/(2*14)
```
Chao1 Richness estimator of sample: (smaller sample of the community)
```{r}
14 + 1^2/(2*9)
```


####Part 4: Alpha-diversity functions in R
```{r}
library(vegan)
```
```{r}
sample_data1_diversity = 
  sample_data1 %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

sample_data1_diversity
```
```{r}
diversity(sample_data1_diversity, index="invsimpson")
```
```{r}
specpool(sample_data1_diversity)
```
####Part 5: Concluding questions
1) The measure of diversity depends on the definition of species in your samples because diversity estimates are calculated using the number of species that you find. If you use a different species definition when processing your sample (i.e. 98% genetic similarity vs something like 90%) you will get different numbers for your simpson reciprocal index and chao1 richness estimator. This would change your collectors curve too.

2) Yes, you could define species as both the type of candy (which they did) AND the color of the candy. So a blue skittle would be considered a different species than a green skittle. 

3) Different sequencing technologies could influence observed diversity in a sample due to inherent bias when it comes to sequencing the data. For example, if your sequencing technology includes amplifying all sequences using universal primers, you will inevitably skew your data to favor some sequences because there is no such thing as a truly universal primer. 

###Module 3 Writing assignment
Module 3 Writing Assignment: What’s in a [species] name? 

Introduction

In Shakespeare’s tragedy Romeo and Juliet, Juliet famously utters the oft-quoted line “What’s in a name? That which we call a rose, [b]y any other name would smell as sweet”(1). Although this text is not often associated with the concept of taxonomy, Juliet makes a point that can be ascribed to the world beyond the insular society of medieval Verona. Indeed, the concept to which Shakespeare poetically refers has long been a topic of investigation for cognitive psychologists the world over. In the field of cognitive psychology, there is a concept termed “Linguistic-relativism” which hypothesizes that the structure of language directly affects the cognition of those that use it(2). To examine this concept in the words of taxonomy, the fact that we assign specific names and taxonomic identifications to genetically distinct organisms directly affects how we consider those organisms to be different from one another. If we accept this theory to be true, this makes the practice of taxonomic classification especially important. Taxonomy was first conceived as a way of classifying multicellular organisms based upon shared characteristics. While this is a relatively easy proposition when one is separating, say, a zebra from a muskrat, it becomes trickier when one ventures into the world of microbes. Should we try to force a multicellular paradigm to fit the prokaryotic  biosphere? That is to say, should microbes be classified into species? Although we need some way of separating microbes into groups based upon genetic similarity, it is debatable whether the established taxonomic system is the best way to do so. The taxonomic system as it exists now fails to accurately depict the full extent of microbial diversity, largely due to the phenomenon of horizontal gene transfer. 

Microbial species definitions

At a multicellular level, species are often defined as a group of organisms with similar physical characteristics which can breed and produce fertile offspring. This definition must be modified when it is applied to microbes, which typically neither have observable physical characteristics, nor do they “breed” in the traditional sense. The rapid asexual division performed by prokaryotes ensures a pool of genetic material which vastly outstrips that of multicellular organisms in terms of sheer diversity. Thus, in order to apply the taxonomic system at a microscopic level, microbial ecologists needed to create a new set of defining characteristics of species. It comes as no surprise, therefore, that the prokaryotic species definition can vary from scientist to scientist. Wayne et al, one of the earliest attempts to establish a bacterial species definition, defines a bacterial species as “a collection of strains that are characterized by at least one diagnostic phenotypic trait and whose purified DNA molecules show at least 70% [DNA] cross-hybridization”(4,5). Other researchers have defined prokaryotic species as sharing 95% or 97% similar rRNA sequences(5). 
Further complicating matters are existing environmental sampling methods for microbial populations. The overwhelming majority of microbial “species” cannot be grown in culture. Thus, the only source of information we have about them are fragments of their genomes that can be isolated from samples of their environments(3). This paucity of data is one of the reasons that the two major bioinformatic pipelines for the analysis of 16S rRNA amplicon data do not classify identified prokaryotes in terms of species, but rather using OTUs and ASVs.

Phylogenetic relationships among prokaryotes

One of the benefits of establishing a taxonomic identification system is that it allows us to infer phylogenetic relationships between different species. As mutations occur at a relatively constant rate, it is only logical that organisms with higher degrees of genetic similarity would be more closely related evolutionarily. While this idea is sound in principle, it overlooks the role of horizontal gene transfer among microbes. Among prokaryotes, genetic information can be taken up from the environment, carried into the cell by a phage, or plasmids can even be transferred directly from cell to cell via a conjugative pilus. Through horizontal gene transfer, entire metabolic pathways have been shared between diverse groups of bacteria and archaea, as has been hypothesized to be the case for specific sulfate respiration pathways(6). Indeed, in the early earth, there is molecular evidence of such promiscuous gene flow that communal evolution was likely the primary method of adaptation(6). 
	
This promiscuous sharing of genetic information has obvious implications when it comes to establishing phylogenies. If two bacteria share specific gene or metabolic pathway, they could be directly evolutionarily related, or merely a product of a horizontal gene transfer that occurred earlier in each respective strain’s history. When drawing a phylogenetic tree of a microbial organism, the lines that represent the inheritance of genetic material don’t just run up and down, but horizontally as well. For this reason, the entire phylogenetic species concept, and the taxonomic system that depends upon it must be restructured to account for this property of the prokaryotic biosphere. 
	
Conclusion

In order to accurately describe the genetic relationships between microbes, we have two options. Either we create a new taxonomic identification system specifically tailored to the microbial world, or we make it understood that the microbial species definition is fundamentally different than that of multicellular organisms. No matter how descriptive a given taxonomic system is, in the end, it is entirely arbitrary. No living species is static, or fits neatly into a taxonomic box. Prokaryotes are merely the most obvious of outliers, due to their twin properties of rapid reproduction and horizontal gene transfer. Even when passaging known species on laboratory media rather than their “natural environment”, large-scale genetic differences can be observed on a human timescale. Bordetella Pertussis, for example, has been found to rearrange significant portions of its genome after as few as 12 passages on laboratory media(7). Imagine how different laboratory adapted Escherichia coli must be, after decades of laboratory maintenance. Consider too, the sheer volume of microbial diversity. Microbes were the first living things to evolve, and have benefited from anywhere between 3.5 and 4 billion years of microbial evolution. Multicellular organisms on the other hand, only evolved around 600 million years ago. It is only to be expected that microbial diversity far outstrips multicellular diversity, commensurate to the billions of extra years it has had to evolve. Thus, to create a clear definition of a microbial species would be an exercise in futility. 

References:
1.	Gibson, R., & Shakespeare, W. (2006). Shakespeare, Romeo and Juliet. Cambridge: Cambridge Univ. Press.
2.	Gleitman, L., & Papafragou, A. (n.d.). Relations Between Language and ought. In Decision Making(pp. 504-523). doi:https://cpb-us-west-2-juc1ugur1qwqqqo4.stackpathdns.com/web.sas.upenn.edu/dist/4/81/files/2017/07/Gleitman-Papafragou-2013_Relations-between-language-and-thought-19a33dc.pdf
3.	Nichols, D. et al. Use of ichip for high-throughput in situ cultivation of “uncultivable” microbial species. Appl. Environ. Microbiol. 76, 2445–2450 (2010)
4.	Wayne L.G, et al. Report of the ad hoc committee on reconciliation of approaches to bacterial systematics. Int. J. Syst. Bacteriol. 1987;37:463–464.
5.	Konstantinidis KT, Ramette A, Tiedje JM. The bacterial species definition in the genomic era. Philosophical Transactions of the Royal Society B: Biological Sciences. 2006;361(1475):1929-1940. doi:10.1098/rstb.2006.1920.
6.	Falkowski, P., Fenchel, T., & Delong, E. (2008). The Microbial Engines That Drive Earth's Biogeochemical Cycles. Science Special Reviews,320.
7.	Brinig, M. M., Cummings, C. A., Sanden, G. N., Stefanelli, P., Lawrence, A., & Relman, D. A. (2006). Significant Gene Order and Expression Differences in Bordetella pertussis Despite Limited Gene Content Variation. Journal of Bacteriology,188(7), 2375-2382. doi:10.1128/jb.188.7.2375-2382.2006


### Module 03 references
Rockstrom J, et al. 2009. "A safe operating space for humanity" Nature.

Welch RA, Burland V, Plunkett G, et al. Extensive mosaic structure revealed by the complete genome sequence of uropathogenic Escherichia coli. Proceedings of the National Academy of Sciences of the United States of America. 2002;99(26):17020-17024. doi:10.1073/pnas.252529799. 

